core/domain/rights_manager.py

```diff
@@ -759,7 +761,9 @@ def _assign_role(
     assignee_username = user_services.get_username(assignee_id)
     old_role = rights_domain.ROLE_NONE
 
-    if new_role == rights_domain.ROLE_OWNER:
+    if allow_assigning_any_role:
```

Comment:
For reviewers: Currently, we have enabled the role removal process for exploration and allowed re-assigning any role to a user (Earlier we were only allowing reassignment to a higher role i.e, from editor to owner but not to playtester.). For now, we are using this flag to enable this feature for expected activity (exploration) and once we will have this feature for other activity (collection) we can remove this flag/option.

Reply:
Could we add a TODO in code to capture this? That's a nice way to tie the code to the tracking issue for the desired behavior.

Reply:
Thanks for the suggestion, @BenHenning!

Currently, rights_manager only handles rights for collection and exploration and at some point (not very soon), we are planning to remove the collection from the codebase so it will get simpler in any case! :)

@Nik-09, Can you please create a GitHub issue to track the "Simplify _assign_role method while removing "collection" from the application? and add a TODO comment: `TODO(#<issue-id>): <description>` here?

Reply:
Done

Reply:
Thanks!